browser/ui/views/toolbar/brave_toolbar_view.cc

```diff
@@ -169,19 +179,89 @@ void BraveToolbarView::Layout() {
     available_width -= avatar_->GetPreferredSize().width();
     available_width -= element_padding;
   }
-  // Don't allow the omnibox to shrink to the point of non-existence, so
-  // subtract its minimum width from the available width to reserve it.
+
+  // Allow the extension actions to take up a share of the available width,
+  // but consider the minimum for the location bar
+  const int location_bar_min_width = location_bar_->GetMinimumSize().width();
   const int browser_actions_width = browser_actions_->GetWidthForMaxWidth(
-      available_width - location_bar_->GetMinimumSize().width());
+      available_width - location_bar_min_width);
   available_width -= browser_actions_width;
-  const int location_bar_width = available_width;
 
+  // Allow the option of the LocationBar having horizontal margin.
+  // With this option, we support a dynamic percentage of margin, with
+  // a maximum width.
+  const bool restrict_location_bar_width = !location_bar_is_wide_.GetValue();
+  const int location_bar_max_width = restrict_location_bar_width
+                                      ? 1080
+                                      : available_width;
+  const auto toolbar_width = width();
+  int location_bar_width = available_width;
+  int location_bar_margin_h = 0;
+  int location_bar_center_offset = 0;
+  if (restrict_location_bar_width) {
+    double location_bar_margin_h_pc = 0.07;
```

Comment:
How about moving codes for calculating horizontal percent and |location_bar_width| to another methods?
I think Layout() method is getting longer and longer and it would be difficult to read this method in the future.

Reply:
This is done now - to a `BraveToolbarView::SetLocationBarBounds` private function